File: 1346.check-if-n-and-its-double-exist.java
/*
 * @lc app=leetcode id=1346 lang=java
 *
 * [1346] Check If N and Its Double Exist
 */

// @lc code=start

import java.util.HashSet;
import java.util.Set;

class Solution {
    public boolean checkIfExist(int[] arr) {
        Set<Integer> set = new HashSet<>();
        for (int n : arr) {
            if (set.contains(n)) return true;
            set.add(n * 2);
            if (n % 2 == 0) set.add(n / 2);
        }
        return false;
    }
}
// @lc code=end

File: 1346.check-if-n-and-its-double-exist.cpp
/*
 * @lc app=leetcode id=1346 lang=cpp
 *
 * [1346] Check If N and Its Double Exist
 */

// @lc code=start
#include <vector>
#include <set>

using namespace std;

class Solution {
public:
    bool checkIfExist(vector<int>& arr) {
        set<int> seen;
        for (const auto &el : arr) {
            if (auto pos = seen.find(el); pos != seen.end()) return true;
            seen.insert(el * 2);
            if (el % 2 == 0) seen.insert(el / 2);
        }
        return false;
    }
};
// @lc code=end
